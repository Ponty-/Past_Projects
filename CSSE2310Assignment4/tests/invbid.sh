#!/bin/bash

if [ $# != 4 ]
then
    echo "Usage: $0 binarydir port resultdir correct_dir" 
    exit 1
fi



bindir=$1
port=$2
res=$3
right=$4
name=$0

expected=0

if [ ! -f $bindir/client499 ]
then
    echo "$bindir/client499 is missing"
    exit 1
fi

if [ ! -f $bindir/serv499 ]
then
    echo "$bindir/serv499 is missing"
    exit 1
fi



$bindir/serv499 $port .. 1.deck &
spid=$!
$bindir/client499 A game $port > $res/$name.a.out 2>$res/$name.a.err < iamoves &
pid1=$!
$bindir/client499 B game $port > $res/$name.b.out 2>$res/$name.b.err < ibmoves &
pid2=$!
$bindir/client499 C game $port > $res/$name.c.out 2>$res/$name.c.err < icmoves &
pid3=$!
$bindir/client499 D game $port > $res/$name.d.out 2>$res/$name.d.err < idmoves &
pid4=$!

wait $pid1 $pid2 $pid3 $pid4 2> /dev/null
kill $spid 2> /dev/null
wait $spid 2> /dev/null

#if [ $result != $expected ]
#then
#    echo "Expected $expected got $result"
#    exit 1
#fi

for v in a b c d
do

diff -q $res/$name.$v.out $right/$name.$v.out > /dev/null
if [ $? != 0 ]
then
   echo "Stdout mismatch for client $v"
   echo "diff $res/$name.$v.out $right/$name.$v.out"
   exit 2
fi

diff -q $res/$name.$v.err $right/$name.$v.err > /dev/null
if [ $? != 0 ]
then
   echo "Stderr mismatch for client $v"
   echo "diff $res/$name.$v.err $right/$name.$v.err"
   exit 3
fi

done

echo OK
exit 0

