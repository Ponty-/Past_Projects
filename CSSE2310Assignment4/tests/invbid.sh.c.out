Info: ..
Info: Team1: A, C
Info: Team2: B, D
S: 9 4
C: K 9 4
D: A Q 8 4 2
H: J 8 5
Info: A bids 7D
Info: B passes
[7D] - Bid (or pass)> [7D] - Bid (or pass)> [7D] - Bid (or pass)> S: 9 4
C: K 9 4
D: A Q 8 4 2
H: J 8 5
Lead> Info: D plays AS
Info: A plays KD
Info: B plays 3D
Info: C won
S: 9 4
C: K 9 4
D: Q 8 4 2
H: J 8 5
Lead> Info: D plays KS
Info: A plays TD
Info: B plays 6D
Info: C won
S: 9 4
C: K 9 4
D: 8 4 2
H: J 8 5
Lead> Info: D plays JS
Info: A plays 5D
Info: B plays 9D
Info: B won
Info: B plays 6S
S: 9 4
C: K 9 4
D: 4 2
H: J 8 5
[S] play> Info: D plays 8S
Info: A plays 7S
Info: C won
S: 4
C: K 9 4
D: 4 2
H: J 8 5
Lead> Info: D plays JC
Info: A plays 3C
Info: B plays 2C
Info: C won
S: 4
C: 9 4
D: 4 2
H: J 8 5
Lead> Info: D plays 3H
Info: A plays 2H
Info: B plays 4H
Info: C won
S: 4
C: 9 4
D: 4 2
H: 8 5
Lead> Info: D plays 3S
Info: A plays QS
Info: B plays TS
Info: A won
Info: A plays 7D
Info: B plays JD
S:
C: 9 4
D: 4 2
H: 8 5
[D] play> Info: D plays 5S
Info: B won
Info: B plays AC
S:
C: 9 4
D: 2
H: 8 5
[C] play> Info: D plays 8C
Info: A plays 7C
Info: B won
Info: B plays TC
S:
C: 4
D: 2
H: 8 5
[C] play> Info: D plays 5C
Info: A plays QC
Info: A won
Info: A plays 6H
Info: B plays 9H
S:
C:
D: 2
H: 8 5
[H] play> Info: D plays 7H
Info: B won
Info: B plays 6C
S:
C:
D: 2
H: 5
[C] play> Info: D plays 2S
Info: A plays AH
Info: A won
Info: A plays TH
Info: B plays KH
S:
C:
D:
H: 5
[H] play> Info: D plays QH
Info: B won
Info: Team 1=-300, Team 2=0
S: 9 4
C: K 9 4
D: A Q 8 4 2
H: J 8 5
Info: A bids 9H
Info: A plays KD
Info: B plays 3D
S: 9 4
C: K 9 4
D: A Q 8 4 2
H: J 8 5
[D] play> Info: D plays 2S
Info: C won
S: 9 4
C: K 9 4
D: Q 8 4 2
H: J 8 5
Lead> Info: D plays QH
Info: A plays 2H
Info: B plays 4H
Info: D won
Info: D plays AS
Info: A plays QS
Info: B plays TS
S: 9 4
C: K 9 4
D: Q 8 4 2
H: 8 5
[S] play> Info: D won
Info: D plays KS
Info: A plays 7S
Info: B plays 6S
S: 4
C: K 9 4
D: Q 8 4 2
H: 8 5
[S] play> Info: D won
Info: D plays JS
Info: A plays QC
Info: B plays AC
S:
C: K 9 4
D: Q 8 4 2
H: 8 5
[S] play> Info: D won
Info: D plays 8S
Info: A plays 3C
Info: B plays TC
S:
C: 9 4
D: Q 8 4 2
H: 8 5
[S] play> Info: D won
Info: D plays 5S
Info: A plays 7C
Info: B plays 6C
S:
C: 4
D: Q 8 4 2
H: 8 5
[S] play> Info: D won
Info: D plays 3S
Info: A plays 5D
Info: B plays 2C
S:
C:
D: Q 8 4 2
H: 8 5
[S] play> Info: D won
Info: D plays JC
Info: A plays TD
Info: B plays JD
S:
C:
D: 8 4 2
H: 8 5
[C] play> Info: D won
Info: D plays 8C
Info: A plays 7D
Info: B plays 6D
S:
C:
D: 4 2
H: 8 5
[C] play> Info: D won
Info: D plays 5C
Info: A plays AH
Info: B plays 9D
S:
C:
D: 2
H: 8 5
[C] play> Info: A won
Info: A plays 6H
Info: B plays KH
S:
C:
D:
H: 8 5
[H] play> Info: D plays 7H
Info: B won
Info: B plays 9H
S:
C:
D:
H: 5
[H] play> Info: D plays 3H
Info: A plays TH
Info: A won
Info: Team 1=-600, Team 2=0
Info: Winner is Team 2
